### PR TITLE
federal bank changes post review

### PR DESCRIPTION
**Pull request by mahalakshmi-r**

### CODE REVIEW

File: src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy [hunk 1]
```diff
@@ -71,81 +67,51 @@ RoboArmFlow {
...
-                        Thread.sleep(Integer.parseInt(requestMap.bankObj.retryInterval))
-                    } catch (Exception e) {
-                        logError("Open website error ${count}", e)
-                        takeScreenshot("OPEN_WEBSITE_ERROR")
-                        Thread.sleep(Integer.parseInt(requestMap.bankObj.retryInterval))
-                    }
-                    count = count + 1
-                }
-            }
+            requestMap.bankObj.bankUrl
```
Comment (PinakiBB) [inline src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy]: // Remove line number 79

Comment (Visakan) [inline src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy]: //visakan - remove line number 79

File: src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy [hunk 2]
```diff
@@ -157,129 +123,139 @@ RoboArmFlow {
...
+            def name = GebExtensions.scrapeText(driver: driver, webElement: [selectorAttributes: [id: "HREF_AccountSummaryFG.ACCOUNT_NICKNAME_ARRAY[0]"]])
+            def accntType = GebExtensions.scrapeText(driver: driver, webElement: [selectorAttributes: [id: "AccountSummaryFG.ACCOUNT_TYPE_ARRAY[0]"]])
+            def availableBalance = GebExtensions.scrapeText(driver: driver, webElement: [selectorAttributes: [id: "HREF_AccountSummaryFG.BALANCE_ARRAY[0]"]])
+            availableBalance = availableBalance.replace(" Cr.", "").replace(",", "")
+            accountDetails.accountNumber = accNo
+            accountDetails.name = name
+            accountDetails.accountType = accntType
+            accountDetails.availableBalance = availableBalance
+            def result = accountDetails ? "NONE" : "PROFILE_SCRAPPING_FAILED"
+            [success: true, accountDetails: accountDetails, result: result]
```
Comment (Visakan) [inline src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy]: // visakan - you can directly hard code the result as "NONE". I feel line number 147 is not needed.

File: src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy [hunk 3]
```diff
@@ -288,80 +264,78 @@ RoboArmFlow {
...
-            logInfo("BankStatement result ${result}")
             [success: !bankTransactions.isEmpty(), bankTransactions: bankTransactions, result: result]
         }
     }
 
-
+    ExecuteStep("CombinePdf", mandatory: false) {
+        Run {
+            def result = resultMap.result
+            def state = "NONE"
```
Comment (Visakan) [inline src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy]: // visakan - State cannot be assigned as "NONE". state should be a boolean/"SUCCESS or FAILURE". Kindly change it

File: src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy [hunk 4]
```diff
@@ -288,80 +264,78 @@ RoboArmFlow {
...
-        try {
+        GebExtensions.clickButton(driver: driver, webElement: [selectorAttributes: [class: "profile-dropdown-details"]], waitTimeOut: 10)
+        GebExtensions.clickButton(driver: driver, webElement: [selectorAttributes: [id: "HREF_Logout"]], waitTimeOut: 10)
+    }
+
+
+    ExecuteStep("MovetoTraces") {
+        Allowed {
+            resultMap.success
+        }
```
Comment (Visakan) [inline src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy]: // visakan - Can you let me know why is this block used here?

File: src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy [hunk 5]
```diff
@@ -157,129 +123,139 @@ RoboArmFlow {
...
+                files = waitAndGetDownloadedFiles("*.csv", count, 40)
+                count += 1
+                [success: !files.isEmpty(), files: files, result: result]
+            }
+        }
+        ExecuteStep("DownloadPDF", mandatory: false) {
+            Run {
+                def pdfFiles = []
+                def resultList = ["NONE", "NO_TRANSACTIONS", "FAILURE_DOWNLOADING_BANKSTATEMENT"]
+                def result = "NONE"
```
Comment (Visakan) [inline src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy]: //visakan - Kindly use the result from the resultMap

File: src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy (hunk 4 above)
Comment (Visakan) [inline src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy]: //visakan - why is this executeStep after Cleanup? Can we refactor this?

File: src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy (hunk 1 above)
Comment (mahalakshmi-r) [inline src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy]: Removed

File: src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy (hunk 2 above)
Comment (mahalakshmi-r) [inline src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy]: hard coded as NONE

File: src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy (hunk 3 above)
Comment (mahalakshmi-r) [inline src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy]: Initialised as FAILURE

File: src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy (hunk 4 above)
Comment (mahalakshmi-r) [inline src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy]: HDFC is a ref script where pinaki has made this chages

File: src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy (hunk 5 above)
Comment (mahalakshmi-r) [inline src/main/resources/scripts/FEDERAL_BANK_BANKSTATEMENT.groovy]: used resultMap

**Pull request closed without merging**